rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -1467,6 +1509,10 @@ rmw_create_client(
   free_rmw_client.cancel();
   free_service_name.cancel();
 
+  if (TRACETOOLS_TRACEPOINT_ENABLED(rmw_client_init)) {
+    auto gid = client_data->copy_gid();
+    TRACETOOLS_DO_TRACEPOINT(rmw_client_init, static_cast<const void *>(rmw_client), gid.data());
+  }
```

Comment:
Sorry, one more thing.  Since we've simplified this enough now, can this be:

```
TRACETOOLS_TRACEPOINT(
  rmw_client_init,
  static_cast<const void *>(rmw_client),
  client_data->copy_gid().data());
```

(I ended up noticing it because `gid` ends up being unused when tracing is compiled out)

Reply:
changed!

`TRACETOOLS_TRACEPOINT_ENABLED()` should be defined to `false` when `TRACETOOLS_DISABLED`, not sure why it's not optimized out

Reply:
> `TRACETOOLS_TRACEPOINT_ENABLED()` should be defined to `false` when `TRACETOOLS_DISABLED`, not sure why it's not optimized out

Ah, that's because I was building with `-DCMAKE_BUILD_TYPE=Debug`.  Anyway, this is shorter now and accomplishes the same thing.